matchers/match_error_matcher.go

```diff
@@ -22,8 +21,9 @@ func (matcher *MatchErrorMatcher) Match(actual interface{}) (success bool, err e
 
 	actualErr := actual.(error)
 
-	if isError(matcher.Expected) {
-		return reflect.DeepEqual(actualErr, matcher.Expected), nil
+	// deepEqual doesn't match errors anymore
+	if expectedErr, ok := matcher.Expected.(error); ok {
```

Comment:
I think in this block we can possibly also compare underlying types of the errors.